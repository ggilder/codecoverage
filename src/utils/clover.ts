import * as clover from '@cvrg-report/clover-json'
import * as fs from 'fs'
import * as path from 'path'
import {CoverageParsed} from './general'

export async function parseClover(
  coveragePath: string,
  workspacePath: string
): Promise<CoverageParsed> {
  if (!coveragePath) {
    throw Error('No Clover XML path provided')
  }
  if (!workspacePath) {
    throw Error('No workspace path provided')
  }
  const fileRaw = fs.readFileSync(coveragePath, 'utf8')
  const parsed = await clover.parseContent(fileRaw)

  for (const entry of parsed) {
    entry.file = path.relative(workspacePath, entry.path || entry.file)
  }

  return parsed
}
