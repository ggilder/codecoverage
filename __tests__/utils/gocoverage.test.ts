import {test} from '@jest/globals'
import {parseGoCoverage} from '../../src/utils/gocoverage'
import {getFixturePath} from '../fixtures/util'

test('should parse Go coverage file', async function () {
  const path = getFixturePath('gocoverage.out')
  const goModPath = getFixturePath('go.mod')
  const output = await parseGoCoverage(path, goModPath)
  expect(output).toMatchSnapshot()
})

test('should throw err if file path is not given', async function () {
  await expect(parseGoCoverage('', '')).rejects.toThrow(
    'No Go coverage path provided'
  )
})

test('should throw err if go.mod path is not given', async function () {
  await expect(parseGoCoverage('foo', '')).rejects.toThrow(
    'No Go module path provided'
  )
})
