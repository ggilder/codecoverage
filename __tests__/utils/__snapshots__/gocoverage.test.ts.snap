// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should parse Go coverage file 1`] = `
[
  {
    "file": "bitrot.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 57,
        },
        {
          "hit": 1,
          "line": 58,
        },
        {
          "hit": 1,
          "line": 59,
        },
        {
          "hit": 0,
          "line": 60,
        },
        {
          "hit": 0,
          "line": 61,
        },
        {
          "hit": 1,
          "line": 62,
        },
        {
          "hit": 1,
          "line": 65,
        },
        {
          "hit": 1,
          "line": 66,
        },
        {
          "hit": 1,
          "line": 67,
        },
        {
          "hit": 0,
          "line": 68,
        },
        {
          "hit": 0,
          "line": 69,
        },
        {
          "hit": 1,
          "line": 70,
        },
        {
          "hit": 1,
          "line": 71,
        },
        {
          "hit": 1,
          "line": 72,
        },
        {
          "hit": 0,
          "line": 73,
        },
        {
          "hit": 0,
          "line": 74,
        },
        {
          "hit": 1,
          "line": 75,
        },
        {
          "hit": 1,
          "line": 76,
        },
        {
          "hit": 1,
          "line": 77,
        },
        {
          "hit": 1,
          "line": 78,
        },
        {
          "hit": 1,
          "line": 79,
        },
        {
          "hit": 1,
          "line": 80,
        },
        {
          "hit": 0,
          "line": 81,
        },
        {
          "hit": 0,
          "line": 82,
        },
        {
          "hit": 1,
          "line": 85,
        },
        {
          "hit": 1,
          "line": 86,
        },
        {
          "hit": 0,
          "line": 87,
        },
        {
          "hit": 0,
          "line": 88,
        },
        {
          "hit": 2,
          "line": 89,
        },
        {
          "hit": 1,
          "line": 90,
        },
        {
          "hit": 1,
          "line": 91,
        },
        {
          "hit": 1,
          "line": 92,
        },
        {
          "hit": 1,
          "line": 93,
        },
        {
          "hit": 1,
          "line": 94,
        },
        {
          "hit": 1,
          "line": 95,
        },
        {
          "hit": 1,
          "line": 98,
        },
        {
          "hit": 1,
          "line": 99,
        },
        {
          "hit": 0,
          "line": 100,
        },
        {
          "hit": 0,
          "line": 101,
        },
        {
          "hit": 0,
          "line": 102,
        },
        {
          "hit": 1,
          "line": 104,
        },
        {
          "hit": 1,
          "line": 105,
        },
        {
          "hit": 1,
          "line": 106,
        },
        {
          "hit": 1,
          "line": 109,
        },
        {
          "hit": 1,
          "line": 110,
        },
        {
          "hit": 1,
          "line": 111,
        },
        {
          "hit": 0,
          "line": 112,
        },
        {
          "hit": 0,
          "line": 113,
        },
        {
          "hit": 1,
          "line": 114,
        },
        {
          "hit": 1,
          "line": 115,
        },
        {
          "hit": 1,
          "line": 116,
        },
        {
          "hit": 0,
          "line": 117,
        },
        {
          "hit": 0,
          "line": 118,
        },
        {
          "hit": 1,
          "line": 119,
        },
        {
          "hit": 1,
          "line": 120,
        },
        {
          "hit": 1,
          "line": 121,
        },
        {
          "hit": 1,
          "line": 122,
        },
        {
          "hit": 1,
          "line": 123,
        },
        {
          "hit": 1,
          "line": 124,
        },
        {
          "hit": 0,
          "line": 125,
        },
        {
          "hit": 0,
          "line": 126,
        },
        {
          "hit": 1,
          "line": 128,
        },
        {
          "hit": 1,
          "line": 129,
        },
        {
          "hit": 0,
          "line": 130,
        },
        {
          "hit": 0,
          "line": 131,
        },
        {
          "hit": 2,
          "line": 133,
        },
        {
          "hit": 1,
          "line": 134,
        },
        {
          "hit": 1,
          "line": 135,
        },
        {
          "hit": 1,
          "line": 136,
        },
        {
          "hit": 1,
          "line": 138,
        },
        {
          "hit": 1,
          "line": 139,
        },
        {
          "hit": 1,
          "line": 140,
        },
        {
          "hit": 1,
          "line": 141,
        },
        {
          "hit": 1,
          "line": 142,
        },
        {
          "hit": 2,
          "line": 143,
        },
        {
          "hit": 1,
          "line": 144,
        },
        {
          "hit": 1,
          "line": 145,
        },
        {
          "hit": 2,
          "line": 146,
        },
        {
          "hit": 1,
          "line": 147,
        },
        {
          "hit": 1,
          "line": 148,
        },
        {
          "hit": 1,
          "line": 150,
        },
        {
          "hit": 1,
          "line": 153,
        },
        {
          "hit": 1,
          "line": 154,
        },
        {
          "hit": 1,
          "line": 155,
        },
        {
          "hit": 0,
          "line": 156,
        },
        {
          "hit": 0,
          "line": 157,
        },
        {
          "hit": 1,
          "line": 158,
        },
        {
          "hit": 1,
          "line": 159,
        },
        {
          "hit": 1,
          "line": 160,
        },
        {
          "hit": 0,
          "line": 161,
        },
        {
          "hit": 0,
          "line": 162,
        },
        {
          "hit": 1,
          "line": 164,
        },
        {
          "hit": 1,
          "line": 165,
        },
        {
          "hit": 0,
          "line": 166,
        },
        {
          "hit": 0,
          "line": 167,
        },
        {
          "hit": 1,
          "line": 169,
        },
        {
          "hit": 1,
          "line": 170,
        },
        {
          "hit": 0,
          "line": 171,
        },
        {
          "hit": 0,
          "line": 172,
        },
        {
          "hit": 1,
          "line": 174,
        },
        {
          "hit": 1,
          "line": 175,
        },
        {
          "hit": 0,
          "line": 176,
        },
        {
          "hit": 0,
          "line": 177,
        },
        {
          "hit": 1,
          "line": 179,
        },
        {
          "hit": 1,
          "line": 180,
        },
        {
          "hit": 1,
          "line": 181,
        },
        {
          "hit": 1,
          "line": 182,
        },
        {
          "hit": 1,
          "line": 183,
        },
        {
          "hit": 2,
          "line": 184,
        },
        {
          "hit": 1,
          "line": 185,
        },
        {
          "hit": 1,
          "line": 186,
        },
        {
          "hit": 2,
          "line": 187,
        },
        {
          "hit": 1,
          "line": 188,
        },
        {
          "hit": 1,
          "line": 189,
        },
        {
          "hit": 1,
          "line": 191,
        },
        {
          "hit": 1,
          "line": 194,
        },
        {
          "hit": 1,
          "line": 195,
        },
        {
          "hit": 1,
          "line": 196,
        },
        {
          "hit": 0,
          "line": 197,
        },
        {
          "hit": 0,
          "line": 198,
        },
        {
          "hit": 1,
          "line": 199,
        },
        {
          "hit": 1,
          "line": 200,
        },
        {
          "hit": 1,
          "line": 201,
        },
        {
          "hit": 1,
          "line": 202,
        },
        {
          "hit": 1,
          "line": 203,
        },
        {
          "hit": 0,
          "line": 204,
        },
        {
          "hit": 0,
          "line": 205,
        },
        {
          "hit": 1,
          "line": 207,
        },
        {
          "hit": 1,
          "line": 208,
        },
        {
          "hit": 0,
          "line": 209,
        },
        {
          "hit": 0,
          "line": 210,
        },
        {
          "hit": 1,
          "line": 212,
        },
        {
          "hit": 1,
          "line": 213,
        },
        {
          "hit": 0,
          "line": 214,
        },
        {
          "hit": 0,
          "line": 215,
        },
        {
          "hit": 1,
          "line": 217,
        },
        {
          "hit": 1,
          "line": 218,
        },
        {
          "hit": 0,
          "line": 219,
        },
        {
          "hit": 0,
          "line": 220,
        },
        {
          "hit": 2,
          "line": 222,
        },
        {
          "hit": 1,
          "line": 223,
        },
        {
          "hit": 1,
          "line": 224,
        },
        {
          "hit": 1,
          "line": 225,
        },
        {
          "hit": 1,
          "line": 226,
        },
        {
          "hit": 0,
          "line": 227,
        },
        {
          "hit": 0,
          "line": 228,
        },
        {
          "hit": 0,
          "line": 229,
        },
        {
          "hit": 1,
          "line": 231,
        },
        {
          "hit": 1,
          "line": 232,
        },
        {
          "hit": 1,
          "line": 233,
        },
        {
          "hit": 1,
          "line": 234,
        },
        {
          "hit": 1,
          "line": 235,
        },
        {
          "hit": 1,
          "line": 236,
        },
        {
          "hit": 0,
          "line": 237,
        },
        {
          "hit": 0,
          "line": 238,
        },
        {
          "hit": 1,
          "line": 239,
        },
        {
          "hit": 1,
          "line": 240,
        },
        {
          "hit": 1,
          "line": 241,
        },
        {
          "hit": 1,
          "line": 243,
        },
        {
          "hit": 1,
          "line": 246,
        },
        {
          "hit": 1,
          "line": 247,
        },
        {
          "hit": 0,
          "line": 248,
        },
        {
          "hit": 0,
          "line": 249,
        },
        {
          "hit": 0,
          "line": 252,
        },
        {
          "hit": 0,
          "line": 253,
        },
        {
          "hit": 0,
          "line": 254,
        },
        {
          "hit": 0,
          "line": 255,
        },
        {
          "hit": 0,
          "line": 259,
        },
        {
          "hit": 0,
          "line": 260,
        },
        {
          "hit": 0,
          "line": 261,
        },
        {
          "hit": 0,
          "line": 262,
        },
        {
          "hit": 0,
          "line": 263,
        },
        {
          "hit": 0,
          "line": 264,
        },
        {
          "hit": 0,
          "line": 265,
        },
        {
          "hit": 0,
          "line": 266,
        },
        {
          "hit": 0,
          "line": 267,
        },
        {
          "hit": 0,
          "line": 268,
        },
        {
          "hit": 0,
          "line": 269,
        },
        {
          "hit": 0,
          "line": 270,
        },
        {
          "hit": 0,
          "line": 271,
        },
        {
          "hit": 0,
          "line": 272,
        },
        {
          "hit": 0,
          "line": 273,
        },
        {
          "hit": 0,
          "line": 274,
        },
        {
          "hit": 0,
          "line": 275,
        },
        {
          "hit": 0,
          "line": 276,
        },
        {
          "hit": 0,
          "line": 277,
        },
        {
          "hit": 0,
          "line": 278,
        },
        {
          "hit": 0,
          "line": 279,
        },
        {
          "hit": 0,
          "line": 280,
        },
        {
          "hit": 0,
          "line": 281,
        },
        {
          "hit": 0,
          "line": 282,
        },
        {
          "hit": 0,
          "line": 283,
        },
        {
          "hit": 0,
          "line": 284,
        },
        {
          "hit": 0,
          "line": 285,
        },
        {
          "hit": 0,
          "line": 286,
        },
        {
          "hit": 0,
          "line": 287,
        },
        {
          "hit": 0,
          "line": 288,
        },
        {
          "hit": 0,
          "line": 289,
        },
        {
          "hit": 0,
          "line": 290,
        },
        {
          "hit": 0,
          "line": 291,
        },
        {
          "hit": 0,
          "line": 292,
        },
        {
          "hit": 0,
          "line": 293,
        },
        {
          "hit": 0,
          "line": 294,
        },
        {
          "hit": 0,
          "line": 295,
        },
        {
          "hit": 0,
          "line": 296,
        },
        {
          "hit": 0,
          "line": 297,
        },
        {
          "hit": 0,
          "line": 298,
        },
        {
          "hit": 0,
          "line": 299,
        },
        {
          "hit": 0,
          "line": 300,
        },
        {
          "hit": 0,
          "line": 301,
        },
        {
          "hit": 0,
          "line": 302,
        },
        {
          "hit": 0,
          "line": 303,
        },
      ],
      "found": 247,
      "hit": 115,
    },
    "title": "bitrot.go",
  },
  {
    "file": "checksum.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 16,
        },
        {
          "hit": 1,
          "line": 17,
        },
        {
          "hit": 1,
          "line": 18,
        },
        {
          "hit": 0,
          "line": 19,
        },
        {
          "hit": 0,
          "line": 20,
        },
        {
          "hit": 1,
          "line": 21,
        },
        {
          "hit": 1,
          "line": 22,
        },
        {
          "hit": 1,
          "line": 23,
        },
        {
          "hit": 1,
          "line": 24,
        },
        {
          "hit": 1,
          "line": 25,
        },
        {
          "hit": 1,
          "line": 28,
        },
        {
          "hit": 1,
          "line": 29,
        },
        {
          "hit": 1,
          "line": 30,
        },
        {
          "hit": 0,
          "line": 31,
        },
        {
          "hit": 0,
          "line": 32,
        },
        {
          "hit": 1,
          "line": 33,
        },
        {
          "hit": 1,
          "line": 34,
        },
        {
          "hit": 0,
          "line": 35,
        },
        {
          "hit": 0,
          "line": 36,
        },
        {
          "hit": 1,
          "line": 37,
        },
        {
          "hit": 1,
          "line": 40,
        },
        {
          "hit": 2,
          "line": 41,
        },
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 44,
        },
        {
          "hit": 0,
          "line": 45,
        },
        {
          "hit": 0,
          "line": 46,
        },
        {
          "hit": 1,
          "line": 47,
        },
        {
          "hit": 2,
          "line": 48,
        },
        {
          "hit": 1,
          "line": 49,
        },
        {
          "hit": 1,
          "line": 50,
        },
      ],
      "found": 37,
      "hit": 23,
    },
    "title": "checksum.go",
  },
  {
    "file": "comparison_report.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 12,
        },
        {
          "hit": 1,
          "line": 13,
        },
        {
          "hit": 1,
          "line": 14,
        },
        {
          "hit": 1,
          "line": 15,
        },
        {
          "hit": 1,
          "line": 17,
        },
        {
          "hit": 1,
          "line": 18,
        },
        {
          "hit": 1,
          "line": 19,
        },
        {
          "hit": 1,
          "line": 21,
        },
        {
          "hit": 1,
          "line": 22,
        },
        {
          "hit": 2,
          "line": 23,
        },
        {
          "hit": 1,
          "line": 24,
        },
        {
          "hit": 2,
          "line": 25,
        },
        {
          "hit": 1,
          "line": 26,
        },
        {
          "hit": 1,
          "line": 27,
        },
        {
          "hit": 1,
          "line": 28,
        },
        {
          "hit": 1,
          "line": 29,
        },
        {
          "hit": 1,
          "line": 30,
        },
        {
          "hit": 1,
          "line": 31,
        },
        {
          "hit": 1,
          "line": 32,
        },
        {
          "hit": 1,
          "line": 33,
        },
        {
          "hit": 1,
          "line": 34,
        },
        {
          "hit": 1,
          "line": 35,
        },
        {
          "hit": 1,
          "line": 36,
        },
        {
          "hit": 1,
          "line": 37,
        },
        {
          "hit": 1,
          "line": 38,
        },
        {
          "hit": 1,
          "line": 39,
        },
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 44,
        },
        {
          "hit": 1,
          "line": 45,
        },
        {
          "hit": 1,
          "line": 46,
        },
        {
          "hit": 1,
          "line": 47,
        },
        {
          "hit": 1,
          "line": 48,
        },
        {
          "hit": 1,
          "line": 49,
        },
        {
          "hit": 1,
          "line": 51,
        },
        {
          "hit": 1,
          "line": 52,
        },
        {
          "hit": 1,
          "line": 53,
        },
        {
          "hit": 1,
          "line": 54,
        },
        {
          "hit": 1,
          "line": 56,
        },
        {
          "hit": 1,
          "line": 57,
        },
        {
          "hit": 2,
          "line": 58,
        },
        {
          "hit": 1,
          "line": 59,
        },
        {
          "hit": 1,
          "line": 60,
        },
        {
          "hit": 1,
          "line": 61,
        },
        {
          "hit": 1,
          "line": 64,
        },
        {
          "hit": 1,
          "line": 65,
        },
        {
          "hit": 1,
          "line": 66,
        },
        {
          "hit": 1,
          "line": 68,
        },
        {
          "hit": 1,
          "line": 69,
        },
        {
          "hit": 1,
          "line": 70,
        },
        {
          "hit": 1,
          "line": 71,
        },
        {
          "hit": 1,
          "line": 72,
        },
        {
          "hit": 2,
          "line": 73,
        },
        {
          "hit": 1,
          "line": 74,
        },
        {
          "hit": 1,
          "line": 75,
        },
        {
          "hit": 1,
          "line": 76,
        },
      ],
      "found": 60,
      "hit": 56,
    },
    "title": "comparison_report.go",
  },
  {
    "file": "config.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 44,
        },
        {
          "hit": 0,
          "line": 45,
        },
        {
          "hit": 0,
          "line": 46,
        },
        {
          "hit": 0,
          "line": 47,
        },
        {
          "hit": 0,
          "line": 48,
        },
        {
          "hit": 1,
          "line": 51,
        },
        {
          "hit": 1,
          "line": 52,
        },
        {
          "hit": 1,
          "line": 53,
        },
        {
          "hit": 1,
          "line": 54,
        },
        {
          "hit": 1,
          "line": 57,
        },
        {
          "hit": 1,
          "line": 58,
        },
        {
          "hit": 2,
          "line": 59,
        },
        {
          "hit": 2,
          "line": 60,
        },
        {
          "hit": 1,
          "line": 61,
        },
        {
          "hit": 1,
          "line": 62,
        },
        {
          "hit": 1,
          "line": 64,
        },
        {
          "hit": 1,
          "line": 67,
        },
        {
          "hit": 2,
          "line": 68,
        },
        {
          "hit": 1,
          "line": 69,
        },
        {
          "hit": 1,
          "line": 70,
        },
        {
          "hit": 1,
          "line": 71,
        },
      ],
      "found": 28,
      "hit": 19,
    },
    "title": "config.go",
  },
  {
    "file": "manifest.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 26,
        },
        {
          "hit": 1,
          "line": 27,
        },
        {
          "hit": 1,
          "line": 28,
        },
        {
          "hit": 0,
          "line": 29,
        },
        {
          "hit": 0,
          "line": 30,
        },
        {
          "hit": 1,
          "line": 32,
        },
        {
          "hit": 1,
          "line": 33,
        },
        {
          "hit": 1,
          "line": 34,
        },
        {
          "hit": 1,
          "line": 35,
        },
        {
          "hit": 1,
          "line": 36,
        },
        {
          "hit": 1,
          "line": 41,
        },
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 44,
        },
        {
          "hit": 1,
          "line": 45,
        },
        {
          "hit": 0,
          "line": 46,
        },
        {
          "hit": 0,
          "line": 47,
        },
        {
          "hit": 1,
          "line": 48,
        },
        {
          "hit": 1,
          "line": 49,
        },
        {
          "hit": 0,
          "line": 50,
        },
        {
          "hit": 0,
          "line": 51,
        },
        {
          "hit": 1,
          "line": 52,
        },
        {
          "hit": 0,
          "line": 55,
        },
        {
          "hit": 0,
          "line": 56,
        },
        {
          "hit": 0,
          "line": 57,
        },
        {
          "hit": 0,
          "line": 58,
        },
        {
          "hit": 1,
          "line": 60,
        },
        {
          "hit": 1,
          "line": 61,
        },
        {
          "hit": 2,
          "line": 62,
        },
        {
          "hit": 1,
          "line": 63,
        },
        {
          "hit": 0,
          "line": 64,
        },
        {
          "hit": 0,
          "line": 65,
        },
        {
          "hit": 2,
          "line": 67,
        },
        {
          "hit": 2,
          "line": 68,
        },
        {
          "hit": 1,
          "line": 69,
        },
        {
          "hit": 1,
          "line": 70,
        },
        {
          "hit": 1,
          "line": 71,
        },
        {
          "hit": 1,
          "line": 72,
        },
        {
          "hit": 2,
          "line": 75,
        },
        {
          "hit": 1,
          "line": 76,
        },
        {
          "hit": 1,
          "line": 77,
        },
        {
          "hit": 1,
          "line": 78,
        },
        {
          "hit": 0,
          "line": 79,
        },
        {
          "hit": 0,
          "line": 80,
        },
        {
          "hit": 1,
          "line": 82,
        },
        {
          "hit": 1,
          "line": 83,
        },
        {
          "hit": 1,
          "line": 84,
        },
        {
          "hit": 0,
          "line": 85,
        },
        {
          "hit": 0,
          "line": 86,
        },
        {
          "hit": 1,
          "line": 87,
        },
        {
          "hit": 1,
          "line": 88,
        },
        {
          "hit": 1,
          "line": 89,
        },
        {
          "hit": 1,
          "line": 90,
        },
        {
          "hit": 1,
          "line": 93,
        },
        {
          "hit": 1,
          "line": 95,
        },
        {
          "hit": 0,
          "line": 96,
        },
        {
          "hit": 0,
          "line": 97,
        },
        {
          "hit": 1,
          "line": 98,
        },
      ],
      "found": 69,
      "hit": 40,
    },
    "title": "manifest.go",
  },
  {
    "file": "manifest_comparison.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 26,
        },
        {
          "hit": 1,
          "line": 27,
        },
        {
          "hit": 1,
          "line": 28,
        },
        {
          "hit": 1,
          "line": 29,
        },
        {
          "hit": 1,
          "line": 30,
        },
        {
          "hit": 1,
          "line": 32,
        },
        {
          "hit": 1,
          "line": 33,
        },
        {
          "hit": 1,
          "line": 34,
        },
        {
          "hit": 1,
          "line": 36,
        },
        {
          "hit": 1,
          "line": 37,
        },
        {
          "hit": 1,
          "line": 38,
        },
        {
          "hit": 1,
          "line": 39,
        },
        {
          "hit": 1,
          "line": 40,
        },
        {
          "hit": 1,
          "line": 41,
        },
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 45,
        },
        {
          "hit": 1,
          "line": 46,
        },
        {
          "hit": 2,
          "line": 47,
        },
        {
          "hit": 1,
          "line": 48,
        },
        {
          "hit": 1,
          "line": 49,
        },
        {
          "hit": 2,
          "line": 52,
        },
        {
          "hit": 1,
          "line": 53,
        },
        {
          "hit": 2,
          "line": 54,
        },
        {
          "hit": 1,
          "line": 55,
        },
        {
          "hit": 1,
          "line": 56,
        },
        {
          "hit": 2,
          "line": 60,
        },
        {
          "hit": 1,
          "line": 61,
        },
        {
          "hit": 2,
          "line": 62,
        },
        {
          "hit": 1,
          "line": 63,
        },
        {
          "hit": 2,
          "line": 67,
        },
        {
          "hit": 1,
          "line": 68,
        },
        {
          "hit": 1,
          "line": 72,
        },
        {
          "hit": 1,
          "line": 75,
        },
        {
          "hit": 1,
          "line": 78,
        },
        {
          "hit": 1,
          "line": 79,
        },
        {
          "hit": 2,
          "line": 80,
        },
        {
          "hit": 1,
          "line": 81,
        },
        {
          "hit": 1,
          "line": 82,
        },
        {
          "hit": 2,
          "line": 84,
        },
        {
          "hit": 1,
          "line": 85,
        },
        {
          "hit": 2,
          "line": 86,
        },
        {
          "hit": 2,
          "line": 87,
        },
        {
          "hit": 1,
          "line": 88,
        },
        {
          "hit": 1,
          "line": 89,
        },
        {
          "hit": 2,
          "line": 90,
        },
        {
          "hit": 1,
          "line": 91,
        },
        {
          "hit": 1,
          "line": 92,
        },
        {
          "hit": 1,
          "line": 93,
        },
        {
          "hit": 1,
          "line": 96,
        },
        {
          "hit": 1,
          "line": 99,
        },
        {
          "hit": 1,
          "line": 100,
        },
        {
          "hit": 2,
          "line": 101,
        },
        {
          "hit": 1,
          "line": 102,
        },
        {
          "hit": 1,
          "line": 103,
        },
        {
          "hit": 1,
          "line": 105,
        },
        {
          "hit": 1,
          "line": 106,
        },
        {
          "hit": 1,
          "line": 107,
        },
        {
          "hit": 2,
          "line": 108,
        },
        {
          "hit": 2,
          "line": 109,
        },
        {
          "hit": 1,
          "line": 110,
        },
        {
          "hit": 1,
          "line": 111,
        },
        {
          "hit": 1,
          "line": 115,
        },
        {
          "hit": 1,
          "line": 118,
        },
        {
          "hit": 2,
          "line": 119,
        },
        {
          "hit": 2,
          "line": 120,
        },
        {
          "hit": 1,
          "line": 121,
        },
        {
          "hit": 1,
          "line": 122,
        },
        {
          "hit": 1,
          "line": 124,
        },
      ],
      "found": 85,
      "hit": 69,
    },
    "title": "manifest_comparison.go",
  },
  {
    "file": "manifest_storage.go",
    "lines": {
      "details": [
        {
          "hit": 1,
          "line": 42,
        },
        {
          "hit": 1,
          "line": 43,
        },
        {
          "hit": 1,
          "line": 44,
        },
        {
          "hit": 1,
          "line": 46,
        },
        {
          "hit": 1,
          "line": 47,
        },
        {
          "hit": 1,
          "line": 48,
        },
        {
          "hit": 2,
          "line": 49,
        },
        {
          "hit": 1,
          "line": 50,
        },
        {
          "hit": 1,
          "line": 51,
        },
        {
          "hit": 0,
          "line": 52,
        },
        {
          "hit": 0,
          "line": 53,
        },
        {
          "hit": 1,
          "line": 55,
        },
        {
          "hit": 1,
          "line": 56,
        },
        {
          "hit": 1,
          "line": 57,
        },
        {
          "hit": 1,
          "line": 58,
        },
        {
          "hit": 1,
          "line": 61,
        },
        {
          "hit": 1,
          "line": 64,
        },
        {
          "hit": 1,
          "line": 65,
        },
        {
          "hit": 1,
          "line": 66,
        },
        {
          "hit": 0,
          "line": 67,
        },
        {
          "hit": 0,
          "line": 68,
        },
        {
          "hit": 1,
          "line": 70,
        },
        {
          "hit": 1,
          "line": 71,
        },
        {
          "hit": 0,
          "line": 72,
        },
        {
          "hit": 0,
          "line": 73,
        },
        {
          "hit": 1,
          "line": 74,
        },
        {
          "hit": 1,
          "line": 75,
        },
        {
          "hit": 1,
          "line": 76,
        },
        {
          "hit": 2,
          "line": 77,
        },
        {
          "hit": 1,
          "line": 78,
        },
        {
          "hit": 1,
          "line": 79,
        },
        {
          "hit": 0,
          "line": 80,
        },
        {
          "hit": 0,
          "line": 81,
        },
        {
          "hit": 0,
          "line": 82,
        },
        {
          "hit": 0,
          "line": 83,
        },
        {
          "hit": 0,
          "line": 84,
        },
        {
          "hit": 1,
          "line": 86,
        },
        {
          "hit": 1,
          "line": 89,
        },
        {
          "hit": 1,
          "line": 90,
        },
        {
          "hit": 1,
          "line": 91,
        },
        {
          "hit": 0,
          "line": 92,
        },
        {
          "hit": 0,
          "line": 93,
        },
        {
          "hit": 1,
          "line": 95,
        },
        {
          "hit": 1,
          "line": 96,
        },
        {
          "hit": 0,
          "line": 97,
        },
        {
          "hit": 0,
          "line": 98,
        },
        {
          "hit": 1,
          "line": 99,
        },
        {
          "hit": 1,
          "line": 100,
        },
        {
          "hit": 2,
          "line": 101,
        },
        {
          "hit": 1,
          "line": 102,
        },
        {
          "hit": 1,
          "line": 103,
        },
        {
          "hit": 1,
          "line": 104,
        },
        {
          "hit": 1,
          "line": 107,
        },
        {
          "hit": 1,
          "line": 108,
        },
        {
          "hit": 1,
          "line": 109,
        },
        {
          "hit": 0,
          "line": 110,
        },
        {
          "hit": 0,
          "line": 111,
        },
        {
          "hit": 1,
          "line": 113,
        },
        {
          "hit": 1,
          "line": 114,
        },
        {
          "hit": 1,
          "line": 115,
        },
        {
          "hit": 0,
          "line": 116,
        },
        {
          "hit": 0,
          "line": 117,
        },
        {
          "hit": 1,
          "line": 118,
        },
        {
          "hit": 1,
          "line": 121,
        },
        {
          "hit": 1,
          "line": 122,
        },
        {
          "hit": 1,
          "line": 123,
        },
        {
          "hit": 1,
          "line": 124,
        },
        {
          "hit": 0,
          "line": 125,
        },
        {
          "hit": 0,
          "line": 126,
        },
        {
          "hit": 1,
          "line": 127,
        },
        {
          "hit": 1,
          "line": 128,
        },
        {
          "hit": 0,
          "line": 129,
        },
        {
          "hit": 0,
          "line": 130,
        },
        {
          "hit": 1,
          "line": 131,
        },
        {
          "hit": 1,
          "line": 134,
        },
        {
          "hit": 1,
          "line": 135,
        },
        {
          "hit": 1,
          "line": 136,
        },
        {
          "hit": 1,
          "line": 137,
        },
        {
          "hit": 1,
          "line": 138,
        },
        {
          "hit": 0,
          "line": 139,
        },
        {
          "hit": 0,
          "line": 140,
        },
        {
          "hit": 1,
          "line": 143,
        },
        {
          "hit": 2,
          "line": 144,
        },
        {
          "hit": 1,
          "line": 145,
        },
        {
          "hit": 1,
          "line": 146,
        },
        {
          "hit": 1,
          "line": 147,
        },
        {
          "hit": 1,
          "line": 148,
        },
        {
          "hit": 0,
          "line": 149,
        },
        {
          "hit": 0,
          "line": 150,
        },
        {
          "hit": 1,
          "line": 151,
        },
        {
          "hit": 1,
          "line": 152,
        },
        {
          "hit": 1,
          "line": 153,
        },
        {
          "hit": 1,
          "line": 154,
        },
        {
          "hit": 1,
          "line": 155,
        },
        {
          "hit": 0,
          "line": 156,
        },
        {
          "hit": 0,
          "line": 157,
        },
        {
          "hit": 1,
          "line": 158,
        },
        {
          "hit": 0,
          "line": 159,
        },
        {
          "hit": 0,
          "line": 160,
        },
        {
          "hit": 1,
          "line": 163,
        },
        {
          "hit": 1,
          "line": 166,
        },
        {
          "hit": 1,
          "line": 167,
        },
        {
          "hit": 1,
          "line": 168,
        },
        {
          "hit": 1,
          "line": 169,
        },
        {
          "hit": 1,
          "line": 170,
        },
        {
          "hit": 1,
          "line": 172,
        },
        {
          "hit": 1,
          "line": 173,
        },
        {
          "hit": 1,
          "line": 174,
        },
        {
          "hit": 1,
          "line": 175,
        },
        {
          "hit": 1,
          "line": 176,
        },
        {
          "hit": 1,
          "line": 177,
        },
        {
          "hit": 1,
          "line": 178,
        },
        {
          "hit": 1,
          "line": 181,
        },
        {
          "hit": 1,
          "line": 182,
        },
        {
          "hit": 1,
          "line": 183,
        },
        {
          "hit": 1,
          "line": 184,
        },
        {
          "hit": 1,
          "line": 185,
        },
        {
          "hit": 1,
          "line": 186,
        },
      ],
      "found": 136,
      "hit": 87,
    },
    "title": "manifest_storage.go",
  },
]
`;
